mantidproject/mantid
# Recatagorised muon fitting functions

:warning: Do not merge before #27377  :warning: 

With the addition of many new muon fitting functions it was asked that they be re-categorised to make finding relevant ones easier.

This splits the fitting functions up into generic, specific and modelling.

It also updates the release notes to list the newly added and categorized functions.

**Description of work.**

**To test:**
  * Open Muon Analysis go to the fitting tab and try to add a function. There should be 3 categories Muon Generic, Muon Specific and Muon Modelling.
  * Try running fits with some of the re-catagorised functions to check that they still work
  * Build the release notes and check that all the links are working correctly. 


Fixes #26827  




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
